### DISCUSSION TITLE
Expected workflow and recommendations

### DISCUSSION THREAD

**Discussion by loki42** [inventree/InvenTree#4530] [Q&A] [answered]
I'm slightly confused about what the correct workflow with inventree should be. I have a collection of parts current on reels in boxes, and a collection of parts on feeders on my pick and place machines. My goal is to clearly see the number of parts left on each feeder, and the number of parts left in the boxes, so I can reorder when required. The boards once finished going through SMT line, go on to selective soldering where PTH parts are inserted and then tested and assembled. I am using assemblies in inventree so an  SMT assembled boards requires 0.25 of a panel in it's BOM, and it's then a component for the selective step, etc.  I assume each reel should be a stock item so I can easily move them to the feeders when loading a feeder. Is this correct? So I order 10 reels of 100nF do I add them 10 times and print labels for them, or teach the existing labels? I am keen to add some validation at this step too, so I can scan to a feeder / slot. The machine has validation built in but it seems pointless to do it in two separate steps. 

My conveyors are connected to the network, as are my machines with CAMX. So can I add test steps for each step in the line? I'm keen to keep information so I can see how different pastes perform, if temperature / humidity is an issue etc. I want the machines to consume parts, which I assume I just do by sending requests with the API when the pick and place machines tell me they consume them via CAMX / SECS GEM. Should the line start a work order then finish the work order when the oven tells me it's left the oven? How should I account for mispicks / rejects?

**Comment by matmair** [comment]
@loki42 that is a pretty involved question and the solution will require some code with the current state of the codebase. If you are up to it and allowed, I would like to turn this into a showcase.
Also could you please number your questions so we can make sure everything is answered?

Short answer: If you are using TCP/IP-based SECS GEM you could just hook into that with a small plugin and consume assemblies in InvenTree once they are processed by the last machine. You could do that also after each step - depending on your equipment either automatically or manually with a small checklist screen (probably needs a stacker there to not bottleneck your line).

**Comment by loki42** [comment]
All the board handling (including destacker and unloader) is connected to
the network.  The camx is all over the network too.

1. I assume each reel should be a stock item so I can easily move them to
the feeders when loading a feeder. Is this correct? So I order 10 reels of
100nF do I add them 10 times and print labels for them, or teach the
existing labels? To do this I go into the stock item,  hit duplicate 9
times. This isn't very elegant so I feel like I'm doing it wrong.

2. I am keen to add some validation at this step too, so I can scan to a
feeder / slot. Scan from box of reels to feeder / slot / lane.  If expected
part equals scanned part move else complain.  I could use a property on the
stock location and a plugin for this?

3. As it moves through the line i can call api to add "test results" so i
can record details against the assemblies and track time,  paste used etc.

4. Should the line start a work order when it leaves the destacker then
finish the work order when the oven tells me it's left the oven and is in
the unloader?

 5. How should I account for mispicks / rejects? Move some parts off a
feeder to a stock location that is marked "destroyed"?

**Reply by matmair** [reply, marked as answer]
Thanks for following the format suggestion.
1) Yes, 1 item per reel. If you bunch the order for them into the system, the items get automatically generated so not too much work

2) You could but the question is how that would be communicated best from a UX standpoint.

3) Yes

4) I would start a build order (inventree) the moment you need to make something, then allocate to it all the parts you put on the line on setup. On destack start a new build output (inventree), allocate at appropriate points in the line from the BO to the build output and finish the build before being loaded into the unloader. That way you have an accurate understanding of your material flow and the current state of all builds + overall status of the line (what parts need to be refiled shortly etc).

5) You can mark them in the stock status. I would do a rough count or weigh of the reels/tubes during setup/teardown operations. Depending on your stock levels I would recommend a dedicated area/rack for everything fresh off the line so the line moves at high velocity. The staff can validate levels once the line is at pace again.

**Comment by loki42** [comment]
That all makes sense,  didn't know about 1.

5. I'm still a little confused on.  So I can easily keep track of the
amount of units left on a reel by adjusting when the machine tells me a
part is picked or rejected, however I'm not sure how to record the
rejects.  If I want to use a stock status I'll need a separate stock item
for every part for lost. I guess I could keep it as a part parameter? It's
mostly so I can keep track of parts that are poorly packed or hard to
pick.

**Reply by matmair** [reply]
Docs are not really our strong-suit ATM. Especially introductions.

5. If you want to keep a record of the lost ones you would need to move the stock there. Not sure how it could be done otherwise. Maybe with some sort of plugin. Stock tracking creates an entry when the count is submitted and you can add notes. Maybe that is sufficient enough?
![grafik](https://user-images.githubusercontent.com/66015116/228930015-78c24cc3-550a-4741-9223-ce060ccdc5d9.png)
